### ISSUE TITLE
Add learning rate configuration for optimizers

### ISSUE THREAD

**Issue by kratsg** [scikit-hep/pyhf#986]
# Description

See #741. This would require adding something like 

```
class OptimizerMixin(object):
    def minimize(self, ..., learning_rate=1.0):
        ...
```

and passing this learning rate through the backend shims (probably?) where the gradient may be used depending on `do_grad`. One could consider merging `do_grad` and `learning_rate` into a single variable where `do_grad=True` if `learning_rate > 0.0` and False otherwise (including negative learning rates?).

The shims would simply need to return the gradient multiplied by this learning rate where appropriate

```
return objective, grad * learning_rate
```
